src/boolean.c

```diff
@@ -1123,6 +1123,10 @@ static GSList * boundary_loops (GSList * boundary)
     GSList * inext = i->next;
     GtsSegment * next = inext ? inext->data : start;
     GtsVertex * v = s->v1 == next->v1 || s->v1 == next->v2 ? s->v1 : s->v2;
+    if (!v)
+    {
+      break;
+    }
```

Comment:
you might avoid the crash, but maybe you break too early and do not browse all the segments (e.g. when v == s->v1 == NULL then maybe v should be s->v2 in that case ). 

It is just an idea, I have not tried to understand what this loop is doing.